### ISSUE TITLE
Changing values in a TimesSeriesTableVec3 (Matlab) 

### ISSUE THREAD

**Issue by jimmyDunne** [opensim-org/opensim-core#1218]
I am trying to rotate and alter marker data in TimesSeriesTableVec3. I am able to access to the values of the Vec3's, but not write back to the table. 

```
import org.opensim.modeling.*

%% Use a c3dAdapter to turn read a c3d file
adapter = C3DFileAdapter()

tables = adapter.read('test_static.c3d')

%% get the Markers
markers = tables.get('markers');
%methodsview(markers)
markerlabels = markers.getColumnLabels

%% Define a rotation matix
Rot = 90;          
rotationMatrix = [1,0,0;0,cos(Rot*pi/180),-(sin(Rot*pi/180));0,sin(Rot*pi/180),cos(Rot*pi/180)];
        
%% Rotate the marker locations for each Marker (Column) 
for iMarker = 0 : markers.getNumColumns - 1

    % get the name of the column (string)
    markername = char(markerlabels.get(iMarker));
    % get the column data for the marker
    marker = markers.getDependentColumn(markername);

    % go through each element of the column, rotate the Vec3, and write
    % back to the column data.
    for iRow = 0 : markers.getNumRows - 1
        % get Matlab vector marker position
        vectorData = [marker.getElt(0,iRow).get(0)...
                      marker.getElt(0,iRow).get(1)...
                      marker.getElt(0,iRow).get(2)];

        % rotate the marker data
        rotatedData = [rotationMatrix'*vectorData']';


        %%% Set the rotated data
        %%% I originally used setTo, but that changes the entire column. 
        %%% I need here, a way of writing the markers back to table
        %%%%%%%%%%%%%%%%%%%%%
        marker.setTo(Vec3( rotatedData(1),...
                           rotatedData(2),...
                           rotatedData(3) ) ); 
        %%%%%%%%%%%%%%%%%%%%%                    
                           
                           
    end
end

%% Print the markers to trc file
trcfileadapter = TRCFileAdapter();
trcfileadapter.write(markers,'temp_trc_file_rotated.trc');
```

**Comment by klshrinidhi** [comment]
I think the columns should be retrieved using *~~get~~updDependentColumn* in order to obtain a writable reference:
```
marker = markers.updDependentColumn(markername)
```

**Comment by aseth1** [comment]
I agree with @klshrinidhi but I don't think MATLAB cares about const-ness of references :cry: 

The issue in this block is that marker is a Vector or Vec3. So in the same way you accessed an individual Vec3 and got out its values you need to `set` the values: e.g. `marker.updElt(0,iRow).set(0, rotatedData[0])`  ... I believe.

**Comment by chrisdembia** [comment]
While MATLAB does not know about constness (and SWIG ignores it), it is good for users to use the  **upd** version even in matlab, because there could be side effects.

**Comment by jimmyDunne** [comment]
@aseth1 There is no updElt for VectorViewVec3. 

![image](https://cloud.githubusercontent.com/assets/5296507/18454248/1f529da6-78f8-11e6-8d19-00452d5aa592.png)

**Comment by aseth1** [comment]
>@aseth1 There is no updElt for VectorViewVec3. 

OK. I think you can use `getElt()` to complete the example. We can use this example to discuss a more convenient interface. Even if `updElt()` were available it would still be cumbersome to have to do it this way.

**Comment by jenhicks** [comment]
Need to wrap so there is equivalent updElt() in Matrix, Vector, and RowVector.

**Comment by klshrinidhi** [comment]
This was addressed by #1288. Java tests in that PR test/demonstrate editing rows/columns of DataTable.

**Comment by jenhicks** [comment]
Thanks for the update on this issue @klshrinidhi . I'm going to leave the issue open until @jimmyDunne has a chance to verify in Matlab, as well.

**Comment by aymanhab** [comment]
@jimmyDunne Please let us know if this can be closed by #1288 Thanks.

**Comment by jimmyDunne** [comment]
It can be! thanks.
